src/stdio.jl

```diff
@@ -4,13 +4,42 @@
 
 # logging in verbose mode goes to original stdio streams.  Use macros
 # so that we do not even evaluate the arguments in no-verbose modes
+
+#FYI this could be wrong by -1 second if the now() call happens just before a second boundary and the time() just after
+const utcoffset = floor(Dates.datetime2unix(now())) - floor(time())
+
+function get_log_ts()
+    t = time()
+    ms = t - floor(t) #time() returns utc, now() returns local time
+    dt = Dates.unix2datetime(floor(t+utcoffset))
+    ts = "$(string(dt)[12:end]).$(string(ms)[3:end])"
+end
+
+function get_log_preface()
+    ts = get_log_ts()
+    task = current_task()
+    taskname =
+        if isdefined(IJulia, :tasks)
```

Comment:
Seems like it would be cleaner to use `task_local_storage`.  Have each task call `task_local_storage(:IJulia_task, "myname")` and then in `get_log_preface` use `get(task_local_storage(), :IJulia_task, "")`